lib/api/core/plugins/pluginContext.js

```diff
@@ -157,48 +157,37 @@ class PluginContext {
  * @param {boolean} [overloadProtect]
  * @param {Function} [callback]
  */
-function execute (kuzzle, request, overloadProtect = true, callback = null) {
-  let resolve = null,
-    reject = null, 
-    deferred = null,
-    _callback = callback,
-    _overloadProtect = overloadProtect;
-
-  let error = null;
-  const overloadProtectType = typeof _overloadProtect;
-
-  if (overloadProtectType === 'function') {
-    _callback = _overloadProtect;
-    _overloadProtect = true;
-  }
-  else if (overloadProtectType !== 'boolean') {
-    error = new PluginImplementationError('Invalid argument: the overload protection flag must be a boolean');
-  }
+function execute (kuzzle, request, callback) {
+  let
+    error,
+    resolve,
+    reject,
+    deferred;
 
-  if (!_callback) {
+  if (!callback) {
```

Comment:
As this PR is dealing with the `callback is not a function` errror, I would also add a test before that one to verify that the `callback` argument is, indeed, a callable function, before we start using it in the remaining of this method.

That way we could intercept and raise a proper `PluginImplementationError` instead of letting Kuzzle deal with an uncatched error.

Reply:
done